src/dev/i18n/extract_html_messages.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { jsdom } from 'jsdom';
+import traverse from '@babel/traverse';
+import { parse } from '@babel/parser';
+import {
+  isDirectiveLiteral,
+  isObjectExpression,
+  isStringLiteral,
+  isTemplateLiteral,
+} from '@babel/types';
+
+import { isPropertyWithKey, escapeLineBreak } from './utils';
+import { DEFAULT_MESSAGE_KEY } from './constants';
+
+/**
+ * Find all substrings of "{{ any text }}" pattern
+ */
+const ANGULAR_EXPRESSION_REGEX = /\{\{+([\s\S]*?)\}\}+/g;
+
+/**
+ * Extract default message from an angular filter expression argument
+ * @param {string} expression JavaScript code containing a filter object
+ * @returns {string} Default message
+ */
+function parseFilterObjectExpression(expression) {
+  let message = '';
+  let context;
+
+  // parse an object expression instead of block statement
+  const filterObjectAST = parse(`+${expression}`);
+
+  traverse(filterObjectAST, {
+    enter(path) {
+      if (isObjectExpression(path.node)) {
+        for (const property of path.node.properties) {
+          if (
+            isPropertyWithKey(property, DEFAULT_MESSAGE_KEY) &&
+            isStringLiteral(property.value)
+          ) {
+            message = escapeLineBreak(property.value.value);
+          } else if (
+            isPropertyWithKey(property, DEFAULT_MESSAGE_KEY) &&
+            isTemplateLiteral(property.value)
+          ) {
+            message = escapeLineBreak(property.value.quasis[0].value.raw);
+          } else if (
+            isPropertyWithKey(property, 'context') &&
+            isStringLiteral(property.value)
```

Comment:
__question:__ is there any reason why we allow template literal for default message and don't for context @LeanidShutau? 

__question:__ what is the main reason to support template literals for default messages? New lines?

Reply:
New lines and 'single quotes' without escaping. I think context strings shouldn't be long literals with line breaks or single quotes, context property contains additional information for vendors about message location.
It'll be great, If we can disallow template literals for default messages, because its AST (`property.value.quasis[0].value.raw`) looks too complicated.

Reply:
>  I think context strings shouldn't be long literals with line breaks or single quotes, context property contains additional information for vendors about message location.

Maybe, we'll see :)

> It'll be great, If we can disallow template literals for default messages, because its AST (property.value.quasis[0].value.raw) looks too complicated.

We definitely can if we want, but should throw if we detect template literals in this case so that we don't miss that silently. Up to you, we can leave it as is and see how other developers use it or forbid and return back if people start complaining.